resources/lib/playerinfo.py

```diff
@@ -49,20 +51,28 @@ def onPlayBackStarted(self):  # pylint: disable=invalid-name
         # Get asset_id, title and url from api
         ep_id = play_url_to_id(self.path)
 
-        if ep_id.get('video_id'):
-            episode = self.apihelper.get_episodes(video_id=ep_id.get('video_id'), variety='single')[0]
-        elif ep_id.get('whatson_id'):
-            episode = self.apihelper.get_episodes(whatson_id=ep_id.get('whatson_id'), variety='single')[0]
-        elif ep_id.get('video_url'):
-            episode = self.apihelper.get_episodes(video_url=ep_id.get('video_url'), variety='single')[0]
+        # Get episode data
+        episode = self.get_episode_data(ep_id)
 
         self.asset_id = self.resumepoints.assetpath_to_id(episode.get('assetPath'))
         self.title = episode.get('program')
         self.url = url_to_episode(episode.get('url', ''))
+        self.ep_id = 'S%sE%s' % (episode.get('seasonTitle'), episode.get('episodeNumber'))
         self.whatson_id = episode.get('whatsonId') if episode.get('whatsonId') else None
-
         self.update_position()
-        self.push_position(position=self.last_pos, total=self.total, event='onPlayBackStarted')
+        self.update_total()
+        self.push_position(position=self.last_pos, total=self.total)
+
+    def get_episode_data(self, ep_id):
```

Comment:
This function might as well be an apihelper method, since it only accesses the ApiHelper object.
Although we may want to indicate in the method-description where/how this is used.

Reply:
Okay, I'll add this to apihelper.